src/reanimated2/Hooks.js

```diff
@@ -304,36 +304,41 @@ export function useAnimatedGestureHandler(handlers) {
   return useEvent(
     event => {
       'worklet';
-      const UNDETERMINED = 0;
       const FAILED = 1;
+      const BEGAN = 2;
       const CANCELLED = 3;
       const ACTIVE = 4;
       const END = 5;
 
-      if (event.oldState === UNDETERMINED && handlers.onStart) {
+      if (event.state === BEGAN && handlers.onStart) {
         handlers.onStart(event, context);
       }
       if (event.state === ACTIVE && handlers.onActive) {
         handlers.onActive(event, context);
       }
-      if (event.oldState === ACTIVE && event.state === END && handlers.onEnd) {
+      if (
+        (event.oldState === BEGAN || event.oldState === ACTIVE) &&
```

Comment:
Not sure about this. I don't see any docs regarding this in gesture handler. But I think it should check only for Active. Because onEnd should be triggered after Active.